x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceIndex.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.indices.SystemIndexDescriptor;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+import static org.elasticsearch.index.mapper.MapperService.SINGLE_MAPPING_NAME;
+import static org.elasticsearch.xcontent.XContentFactory.jsonBuilder;
+
+public class InferenceIndex {
+
+    private InferenceIndex() {
+
+    }
+
+    public static final String INDEX_NAME = ".inference";
+    public static final String INDEX_PATTERN = INDEX_NAME + "*";
+
+    // Increment this version number when the mappings change
+    private static final int INDEX_MAPPING_VERSION = 1;
+
+    public static Settings settings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
+            .build();
+    }
+
+    /**
+     * To prevent a mapping explosion dy
```

Comment:
Please finish the comment.

In particular, please say what the strategy will be when we eventually do need to add a new mapping. Most system indices have their mappings upgraded when the whole cluster has been upgraded to the newer version. But for ML system indices we upgrade the mappings more aggressively, when the first action is performed that might create a document with the new fields. We do this for ML because it would be too hard for the C++ processes to be aware of whether they should operate in a compatibility mode because a rolling upgrade is part way through. Maybe this constraint doesn't apply to inference if this system index is only storing configs, and it can adhere to the usual ES approach of only allowing new functionality to be used when the entire cluster is on the version that understands the new mappings.

Reply:
Managing these index mappings is easier than other ml indices as the index is only written to when a new model configuration is PUT, the standard system index functionality can be relied on to ensure correctness on upgrade if new fields are not written until the cluster is fully upgraded. By design `task_settings`, and `service_settings` do not have strict dynamic mappings as the valid options are defined by each service. Only the top level settings are strict and these should not change very often.  

I've filled out the comment and added a test to `ModelRegistryIT` to prove that document that do not comply with the strict mappings are rejected.